openssl/src/x509/mod.rs

```diff
@@ -40,7 +40,63 @@ pub mod extension;
 pub mod store;
 
 #[cfg(test)]
-mod tests;
+mod tests {
+    #[cfg(ossl110)]
+    use x509::X509Builder;
+
+    /// Tests `X509Ref::version` happy path.
+    #[cfg(ossl110)]
+    #[test]
+    fn x509_ref_version() {
+        let mut builder = X509Builder::new().unwrap();
+        let expected_version = 2;
+        builder
+            .set_version(expected_version)
+            .expect("Failed to set certificate version");
+        let cert = builder.build();
+        let actual_version = cert.version();
+        assert_eq!(
+            expected_version, actual_version,
+            "Obtained certificate version is incorrect",
+        );
+    }
+
+    /// Tests `X509Ref::version`. Checks case when no version has been set, so a default one is
+    /// returned.
+    #[cfg(ossl110)]
+    #[test]
+    fn x509_ref_version_no_version_set() {
+        let cert = X509Builder::new().unwrap().build();
+        let actual_version = cert.version();
+        assert_eq!(
+            0, actual_version,
+            "Default certificate version is incorrect",
+        );
+    }
+
+    /// Tests `X509Ref::version`. Checks case when there was an attempt to set incorrect version
+    /// before a call to the tested method.
+    ///
+    /// _Note._ As for now this test is disabled since OpenSSL erroneously allows to set any number
+    /// as a certificate version. See corresponding [`issue`].
+    ///
+    /// [`issue`]: https://github.com/openssl/openssl/issues/12138
+    #[cfg(ossl110)]
+    #[test]
+    #[ignore]
+    fn x509_ref_version_incorrect_version_set() {
```

Comment:
Turns out there is a bug in OpenSSL `X509_set_version(X509 *x, long version)` function. It allows to set any number as described in the corresponding [issue](https://github.com/openssl/openssl/issues/12138). Probably it's worth to consider a temporary check in `X509Builder::set_version` and `X509ReqBuilder::set_version` while the mentioned bug is present?

Reply:
I don't think it makes sense to write a test for a case that the underlying library does not handle.

Reply:
Removed this test.